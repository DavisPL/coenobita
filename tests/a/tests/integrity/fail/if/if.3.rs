#![allow(unused_variables)]
#![allow(unused_assignments)]

fn main() {
    #[cnbt::integrity({bin}{bin})]
    let mut x = 5;
    
    // Since the value assigned to `x` is implicitly affected by
    // crates `b` and `c`, this example should fail to type check
    if b::boolean() && c::boolean() {
        x = 6;
    } else {
        x = 7;
    }
}