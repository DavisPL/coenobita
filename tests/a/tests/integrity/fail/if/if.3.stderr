error: expected {root}{root}, found {root}{b,c,root}
  --> tests/integrity/fail/if/if.3.rs:11:13
   |
11 |         x = 6;
   |             ^
